numpy/_core/tests/test_stringdtype.py

```diff
@@ -458,6 +458,23 @@ def test_creation_functions():
     assert np.empty(3, dtype="T")[0] == ""
 
 
+def test_create_with_copy_none(string_list):
+    arr = np.array(string_list, dtype=StringDType())
+    arr_rev = np.array(string_list[::-1], dtype=StringDType())
```

Comment:
Could you add a comment here, like `# Create a copy with the same dtype but its own allocator`